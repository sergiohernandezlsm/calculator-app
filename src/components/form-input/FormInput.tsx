import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export interface Props {
  name: string;
  placeholder: string;
  text: string;
  calculatorHandler: (x: any) => void;
}

const FormInput: React.FC<Props> = (props) => {
  return (
    <Form.Group className="mb-3" controlId="exampleForm.ControlInput1">
      <Row>
        <Col>
          <Form.Label>{props.name}</Form.Label>
        </Col>
      </Row>
      <Row>
        <Col xs={8}>
          <Form.Control onChange={(e) => props.calculatorHandler(e)} type="number" placeholder={props.placeholder} />
        </Col>
        <Col xs={4}>
          <Form.Text className="text-muted">
            {props.text}
          </Form.Text>
        </Col>
      </Row>
    </Form.Group>
  )
}

export default FormInput;