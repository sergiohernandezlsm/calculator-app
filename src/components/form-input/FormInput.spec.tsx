import { shallow } from 'enzyme';
import FormInput, { Props } from './FormInput';

const props: Props = {
  name: 'string',
  placeholder: 'string',
  text: 'string',
  calculatorHandler: (x: any) => { },
}

describe('FormInput Component', () => {
  it('renders <FormInput /> component', () => {
    const wrapper = shallow(<FormInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});