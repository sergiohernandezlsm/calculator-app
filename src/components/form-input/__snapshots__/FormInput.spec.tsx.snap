// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component renders <FormInput /> component 1`] = `ShallowWrapper {}`;
